### ISSUE TITLE
has_node_feat option in config does not seem to affect the model

### ISSUE THREAD

**Issue by NetherNova** [lrjconan/GRAN#12]
Hi, I am trying to run the generator on a dataset with node features.
However, I noticed, the `has_node_feat `option doesn't seem to do anything.

Is there a way to generate graphs with node features?

**Comment by lrjconan** [comment]
Hi, Thanks for your interest! 
I did not consider node feature since the paper is about pure graph generation.
But it is not hard to add the conditioning on node feature.
Specifically, you can modify this [part](https://github.com/lrjconan/GRAN/blob/6c2bb0f9d1ef2fe00dc156b5205e45d74203ba71/model/gran_mixture_bernoulli.py#L217) to use the available node feature instead of the adjacency matrix.

**Comment by Matt-HJ-Bailey** [comment]
Did you have any luck with this, @NetherNova  ? I'm also interested in generating graphs with node features (in my case, 2D vectors of cartesian coordinates).

**Comment by NetherNova** [comment]
> Did you have any luck with this, @NetherNova ? I'm also interested in generating graphs with node features (in my case, 2D vectors of cartesian coordinates).

Hey, no I have moved away from this method. What @lrjconan is suggesting is to use node features as input, but not in the generated output. To actually generate graphs with node features, let's say discrete features, you'd have to put another node classification layer on top of this method and formulate a joint loss function.

**Comment by dpstart** [comment]
If anyone is interested, I am working on this and I now have a working model with an additional feature head and loss term.

**Comment by tathagatv** [comment]
It appears that the optional node and edge attributes provided in input files as mentioned in the [data README](https://github.com/lrjconan/GRAN/tree/master/data/DD#:~:text=There%20are%20OPTIONAL,with%20graph_id%20i) are not used when provided.

In the forward pass, when [GNN message passing](https://github.com/lrjconan/GRAN/blob/6c2bb0f9d1ef2fe00dc156b5205e45d74203ba71/model/gran_mixture_bernoulli.py#L433) is called, node features are computed on  [A_pad](https://github.com/lrjconan/GRAN/blob/6c2bb0f9d1ef2fe00dc156b5205e45d74203ba71/model/gran_mixture_bernoulli.py#L217) which is the adjacency matrix created from [nx.to_numpy_matrix](https://github.com/lrjconan/GRAN/blob/master/dataset/gran_data.py#:~:text=adj_0%20%3D%20np.array(nx.to_numpy_matrix(G))), but the node and edge features get lost there. 

Will be great if someone can confirm this.

**Comment by dpstart** [comment]
@tathagatv that is correct. I have a version with node features as well if you are interested.

**Comment by thesaturdayafternoon** [comment]
@dpstart Could you please send a link to your repository that contains a version with node features? I really appreciate it.
